### PR TITLE
distroless: avoid repeatedly concatenating package info

### PR DESCRIPTION
**Pull request by BenTheElder**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Overwrite instead of concatenating status info to avoid duplicated lines. Fixes scanning with some SBOM tools. 

#### Does this PR introduce a user-facing change?

```release-note
distroless-iptables image will no longer contain repeated contents in `/var/lib/dpkg/status.d/$package` files
```

### CODE REVIEW

**Comment by BenTheElder** [comment]
flake?
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/release/2831/pull-release-test/1605325116026130432
/retest

**Comment by aojea** [comment]
> /home/prow/go/src/k8s.io/release
> level=error msg="./hack/../dependencies.yaml indicates that k8s.gcr.io/build-image/distroless-iptables should be at version v0.1.2, but the following files didn't match: images/build/distroless-iptables/Makefile"
> level=fatal msg="error during command execution: checking local dependencies: Dependencies are not in sync"
> make: *** [Makefile:49: verify-dependencies] Error 1

**Comment by BenTheElder** [comment]
That's a rather silly entry, this doesn't de-duplicate tracking the version, it _duplicates_ it, since it's only controlling the one source file ...

File: dependencies.yaml
Comment (BenTheElder) [inline dependencies.yaml:409]: this class of entry with a single source file is busy work.

**Comment by aojea** [comment]
now it pass and fails a different one?

**Comment by BenTheElder** [comment]
> now it pass and fails a different one?

It flaked this one on the first push, `pull-release-test` is flaking, specifically k8s.io/release/pkg/release TestPublish flakes

Filed https://github.com/kubernetes/release/issues/2833

/retest
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/release/2831/pull-release-test/1605332879217266688

File: dependencies.yaml
Comment (liggitt) [reply dependencies.yaml:409]: also... v0.2.0 isn't even a dependency version, it's a version this repo is _producing_, right?

Comment (BenTheElder) [reply dependencies.yaml:409]: right.

**Comment by BenTheElder** [comment]
We've confirmed that this currently repeatedly writes identical information to the files, because this image build stages the dependencies of multiple different things which have overlapping dependencies. Each appended section is identical because we're not mutating the set of packages at that point.

Still need approval to merge and start working on updating consumers.

/hold cancel

**Review by justaugustus — APPROVED** [review]
Thanks Ben!

**Comment by BenTheElder** [comment]
#2833 
/retest

**Comment by BenTheElder** [comment]
/retest
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/release/2831/pull-release-integration-test/1607794807759966208
??

**Comment by BenTheElder** [comment]
aside: Should consider producing structured junit test results in the tests for this repo so the individual test failures can be easily identified.

`gotestsum` is typically used to do this for Go unit tests for example. test-infra, KIND, Kubernetes/Kubernetes all have implementations folks can borrow from 🙃

**Comment by BenTheElder** [comment]
commented in slack, the error message for https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/release/2831/pull-release-integration-test/1607797984437735424

is massive and the log is difficult, but to my eyes looks unrelated to this PR
/retest

**Comment by BenTheElder** [comment]
@xmudrii pointed out https://github.com/kubernetes/release/pull/2801, failure feels like it might be similar.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Comment by BenTheElder** [comment]
The postsubmit timed out and unfortunately doesn't appear to be in prow.k8s.io for rerunning https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-release-push-image-distroless-iptables/1607917652074827776

https://prow.k8s.io/?repo=kubernetes%2Frelease&type=postsubmit

at a glance: looks like s390x took too long?

**Pull request merged**
